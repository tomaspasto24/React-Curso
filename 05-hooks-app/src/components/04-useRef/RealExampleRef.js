import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

import '../02-useEffect/effects.css';


export const RealExampleRef = () => {
    
    const [show, setShow] = useState(false);

    return (
        <>
            <h1>RealExampleRef</h1>
            <hr />

            {/* Si show es true muestra el componente. */}
            { show && <MultipleCustomHooks />}

            <button
                onClick={ () => {
                    setShow( !show );
                }}
                className='btn btn-primary mt-5'
            >
                Show/Hide
            </button>
        </>
    )
};