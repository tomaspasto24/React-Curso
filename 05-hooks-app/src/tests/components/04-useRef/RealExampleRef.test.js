import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Pruebas sobre <RealExampleRef />.', () => {

    const wrapper = shallow(<RealExampleRef />);

    test('Debe mostrarse correctamente.', () => {

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(false);
    });

    test('Debe mostrar el componente <MultipleCustomHooks />.', () => {

        wrapper.find('.btn').simulate('click');
        
        expect( wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    });

});