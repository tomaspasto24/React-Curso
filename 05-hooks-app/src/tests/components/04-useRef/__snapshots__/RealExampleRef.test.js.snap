// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <RealExampleRef />. Debe mostrarse correctamente. 1`] = `
<Fragment>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
